File: C++/Competitive Programming/HackerRank/bigSorting.cpp
#include <bits/stdc++.h>

using namespace std;

bool migue(string a, string b){
	unsigned long long n1 = stoull(a);
	unsigned long long n2 = stoull(b);
	return n1 > n2;
}

int main(){
	int n;
	cin >> n;
	vector<string> numeros;
	while(n--){
		string in = "";
		cin >> in;
		numeros.push_back(in);

	}
	sort(numeros.begin(), numeros.end(), migue);
	for (int i = numeros.size()-1; i >= 0; i--)
	{
		cout << numeros[i] << "\n";
	}
	return 0;
}

File: C++/Competitive Programming/HackerRank/bigSorting.java
import java.math.BigInteger;
import java.util.*;
import java.*;

class migue implements Comparator<String> {
    public int compare(String x, String y) {
        if (x.length() == y.length()) {
            return x.compareTo(y);
        }

        return x.length() - y.length();
    }
}

public class bigSorting {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String entrada = "";
		int n = 0;
		n = sc.nextInt();
		String[] numeros = new String[n];
		for(int i = 0; i < n; i++){
			entrada = sc.next();
			numeros[i] = entrada;
		}
		Arrays.sort(numeros, new migue());
		for(int i = 0; i < n; i ++){
			System.out.println(numeros[i]);
		}

	}
}